src/components/pages/Dashboard.jsx

```diff
@@ -1,28 +1,32 @@
-import { logout } from "../../util/fetchData";
+import { logout } from "../../util/DataBaseRequests";
 import { useAuth } from "../../AuthProvider";
 import { useNavigate } from "react-router-dom";
+import { useState } from "react";
 
 const Dashboard = () => {
     // temporarily this functionality is here, then it will be moved to the header
+    const [logoutError, setLogoutError] = useState(false);
+    const { logoutUser, userData } = useAuth();
     
-    const { logoutAction, userData } = useAuth();
     const navigate = useNavigate();
-
     const handleLogout = async () => {
-        logoutAction();
-        logout()
-            .then(() => {
-                logoutAction();
+        try {
+            const result = await logout();
+            if (result.status === 200){
+                logoutUser();
                 navigate('/');
-            })
-            .catch(error => {
-                console.log(error);
-            });
+            }
+        } catch (error) {
+            console.log(error);
+            setLogoutError(true);
+        }
+
     };
     return (
         <div className="flex flex-col items-center">
             <h1 className="text-red font-bold text-2xl sm:text-4xl font-spartan uppercase">Dashboard Page</h1>
-            <h2>Welcome, {userData.role} {userData.firstName} {userData.lastName.slice(0, 1)}.</h2>
+            <h2>Welcome, {userData.role} {userData.firstName} {userData.lastName.slice(0,1)}.</h2>
+            {logoutError && <p>Something went wrong. You're still logged in.</p>}
```

Comment:
We are handling errors a few different ways. In `Login` the error message is contained in the error object. Let's be consistent and follow that pattern.